### ISSUE TITLE
Using GraphSAGE to Train on RDF store

### ISSUE THREAD

**Issue by nkiranrao** [williamleif/GraphSAGE#81]
Hi, 

I was looking for a tool to introduce a ML on a knowledge graph stored in RDF store with a SPARQL endpoint. I noticed GraphSAGE can do more or less everything I am looking i.e. identify a possible graph pattern and apply additional label to enrich the knowledge graph. However I am unable to figure out how to use this tool to work on a graph represented in an RDF database with a SPARQL endpoint.

I am new to ML as well as RDF, so any pointers that suggests if its possible or not will be greatly appreciated.

Regards
Kiran

**Comment by RexYing** [comment]
Hi,
I think this is interesting. the part that you need to modify is mainly the minibatch part. You need to first sample nodes from the RDF, and then provide API that extract the (2hop) neighbors of a given node from the RDF database with bfs, and process it as the input to the tensorflow model (i.e. in the form of placeholders in train)
Another thing to note is that your edge could have multiple types (multiple kinds of relations), in which case you might want to modify the model such that different relations are parameterized by different weights.

**Comment by midnitekoder** [comment]
Does the code base support multiple hop layered aggregation as explained in the paper?
